### ISSUE TITLE
Difference between the same stellar yields in different tables

### ISSUE THREAD

**Issue by JinningLianggithub** [NuGrid/NuPyCEE#45]
Hi builders of NuPyCEE,

I am doing some researches using different stellar yields. I find that there is some slight difference between K10 AGB yields from _agb_and_massive_stars_nugrid_K10.txt_ and from other tables using K10 AGB yields. Also, there is some slight difference between NuGrid AGB yields from _agb_and_massive_stars_nugrid_MESAonly_fryer12mix.txt_ and from other tables using NuGrid AGB yields. But they should be the same.  You can see it from the figure I made. 
<img width="787" alt="YieldsVSinitialmass" src="https://user-images.githubusercontent.com/82739127/194235614-a9a6981b-9696-43f2-9df7-61034f3a6e08.png">
It shows the relation between yields fraction scaled by final mass and initial mass. You can see that there are slight difference for AGB yields between the same yields (i.e. K10-1 and K10-2: other tables contain K10 AGB yields and _agb_and_massive_stars_nugrid_K10.txt_ , NuGrid-1 and NuGrid-2:  other tables contain NuGrid AGB yields and _agb_and_massive_stars_nugrid_MESAonly_fryer12mix.txt_ )

Could you explain why?

**Comment by becot85** [comment]
Hello, Thanks for reaching out. Before I can help you, I think I would need
a bit more details. What is the metallicity you are plotting? How are you
extracting the yields data (via read_yields.py, via Sygma, by hand from txt
files,..)?

Also, when you say that the y axis is the yields scaled by final mass, do
you mean you divide the yields of individual elements by the final mass of
the star? I am asking because the Eu yields in your plot is above -3, it
should be lower by orders of magnitude.. Also, I am not sure I understand
what you mean by K10-1 and K10-2, and by NuGrid-1 and NuGrid-2. Would it be
possible to make a plot with a reduced number of lines (e.g. maybe just
showing the NuGrid cases without the K10 investigation), and maybe have a
more descriptive labels for NuGrid-1 and NuGrid-2?

These clarifications will help me answering your question.
Thanks!
Benoit

**Comment by JinningLianggithub** [comment]
Hi Thank you for replying me!

What is the metallicity you are plotting?  -- 0.02 for all stellar yields
How are you extracting the yields data? -- via read_yields.py
When you say that the y axis is the yields scaled by final mass, do you mean you divide the yields of individual elements by the final mass of the star? -- yes
Also, I am not sure I understand what you mean by K10-1 and K10-2, and by NuGrid-1 and NuGrid-2. -- There are many yields tables using K10, for example, _agb_and_massive_stars_K10_K06_0.0HNe.txt_ (K10-1, blue dashdotdoted lines), _agb_and_massive_stars_nugrid_K10.txt_ (K10-2, black dashdotdoted lines). For LIMs yields or AGB yields, these two files should share the same tables.  But that's not the case. And these yeilds table using NuGrid should share the same AGB yields. But _agb_and_massive_stars_nugrid_MESAonly_fryer12mix.txt_ (NuGrid-2, orange dashdotdoted lines) is different from the others using NuGrid AGB yields.

I am not sure if these can help you answer my question. I can explain more if there still are unclear things.

Best wishes,
Jinning

**Comment by becot85** [comment]
Hello, Thanks for your patience and for the extra info. I did some tests on my end, but I do not see the issue you are having. Below is the script I build to test the NuGrid AGB case. With read_yields.py, I extract the C yields for all NuGrid AGB models using various yields table options (including NuGrid-2). As you can see, all yields for each stellar mass are exactly the same.. Things seem to be fine.

Maybe something went wrong in your normalization? Did you use a similar script than me to extract data? How did you track isotope indices and linked them to elemental yields?

If you do further tests, please note that "agb_and_massive_stars_nugrid_K10.txt" does not have the same number of AGB mass grid points than other yields that use K10. But the ones in common should give the same yields.

Hopefully this helps!
Have a nice day,
Benoit

<img width="1121" alt="1" src="https://user-images.githubusercontent.com/15910680/196480845-6174b80c-5fcc-45d4-929e-eb18dd063e34.png">
<img width="1121" alt="2" src="https://user-images.githubusercontent.com/15910680/196480865-20396994-402d-498f-ae1e-f32d5ff57579.png">
<img width="1105" alt="3" src="https://user-images.githubusercontent.com/15910680/196480875-3a2bedad-d159-4dc2-8fc0-d21711f88090.png">
<img width="1105" alt="4" src="https://user-images.githubusercontent.com/15910680/196480888-8ac08664-af07-409f-afb0-670e188ca15f.png">

**Comment by JinningLianggithub** [comment]
Hi, Benoit!
Thank you for your help and reply!

It is true that "agb_and_massive_stars_nugrid_K10.txt" does not have the same number of AGB mass grid points than other yields that use K10. That's why I find K10-1 and K10-2. As for NuGrid yields, I have checked that my results and they are the same as you show me. But I normalized yields by final mass and ry.read_yields_M_Z().get() cannot return correct final mass. I don't know why. That's why I find NuGrid-1 and NuGrid-2. (For NuGrid AGB yields, when M=1.65, Z=0.02, it should return 0.65 Msun instead of 0.99 Msun)
<img width="602" alt="code" src="https://user-images.githubusercontent.com/82739127/199055826-f8e79bec-e389-4fca-97fe-68991d9480f8.png">
<img width="533" alt="table" src="https://user-images.githubusercontent.com/82739127/199055832-3d13352d-c86a-4dc6-83e7-04a3547e4dbc.png">

Have a nice day,
Jinning

**Comment by becot85** [comment]
Thanks a lot Jinning for this. The inconsistency between the M_final values was because the 1.65Msun model at Z=0.02 in the fryer12mix.txt table had the wrong H-1 yield value, so thanks for pointing this out! I just fixed it and pushed the model to master. Note that all other isotopes were ok, only H-1 was affected for that particular model. The reason you were having different a M_final value is because read_yields.py re-calculate the total ejecta mass by summing the isotopic yields (i.e. it does not blindly use the Mfinal value listed in the table).

Note that I also updated all AGB models from the MESAonly_ye table, they were outdated since they were generated before the official NuGrid (Ritter+18) publication. I swapped them for the ones found in the fryer12 tables.

What were the problematic models related to K10?

Have a nice day,
Benoit
